tests/api/test_page.py

```diff
@@ -210,3 +210,52 @@ def test_paginate_pages(user_api_client, page):
     content = get_graphql_content(response)
     pages_data = content['data']['pages']
     assert len(pages_data['edges']) == 2
+
+
+def test_bulk_publish(
+        staff_api_client, page_list_unpublished,
+        permission_manage_pages):
+    page_list = page_list_unpublished
+    assert not any(page.is_published for page in page_list)
+
+    query = """
+        mutation publishManyPages($ids: [ID]!) {
+            pageBulkPublish(ids: $ids) {
+                count
+            }
+        }
+    """
+    variables = {'ids': [
+        graphene.Node.to_global_id('Page', page.id)
+        for page in page_list]}
+    response = staff_api_client.post_graphql(
+        query, variables, permissions=[permission_manage_pages])
+    content = get_graphql_content(response)
+    for page in page_list:
+        page.refresh_from_db()
+    assert content['data']['pageBulkPublish']['count'] == len(page_list)
+    assert all(page.is_published for page in page_list)
+
+
+def test_bulk_unpublish(
+        staff_api_client, page_list,
+        permission_manage_pages):
+    assert all(page.is_published for page in page_list)
+
+    query = """
+        mutation unPublishManyPages($ids: [ID]!) {
+            pageBulkUnpublish(ids: $ids) {
+                count
+            }
+        }
+    """
+    variables = {'ids': [
+        graphene.Node.to_global_id('Page', page.id)
+        for page in page_list]}
+    response = staff_api_client.post_graphql(
+        query, variables, permissions=[permission_manage_pages])
+    content = get_graphql_content(response)
+    for page in page_list:
+        page.refresh_from_db()
```

Comment:
Maybe I am exaggerating but you run refresh 3 times which is equal to 3 db hits. 
You can easily reduce it to one -  `Page.objects.filter(id__in[page_ids])`